.pre-commit-config.yaml

```diff
@@ -229,3 +229,11 @@ repos:
         entry: python scripts/validate_min_versions_in_sync.py
         language: python
         files: ^(ci/deps/actions-.*-minimum_versions\.yaml|pandas/compat/_optional\.py)$
+    -   id: flake8-pyi
+        name: flake8-pyi
+        entry: flake8 --extend-ignore=E301,E302,E305,E701,E704
+        types: [pyi]
+        language: python
+        additional_dependencies:
```

Comment:
Could these be added to `flake8_dependencies` in the flake8 block above and then this line changed to `additional_dependencies: *flake8_dependencies`?

Reply:
I first tried to add `flake8-pyi` to all the other flake8 plugins but the issue is that we need different ignores for pyi files than for normal py files.

I think flake8 supports file-specific ignores. If they support something along the lines `*.pyi: E301,E302,E305,E701,E704` that would be the best approach. I'll look into that later today.